src/plugins/dashboard/public/dashboard_container/embeddable/api/run_save_functions.tsx

```diff
@@ -158,6 +165,7 @@ export function runSaveAs(this: DashboardContainer) {
             this.controlGroup.setSavedState(persistableControlGroupInput);
           }
         });
+        this.updateInput(dashboardStateToSave);
```

Comment:
This is important to handle a race condition when Save as new dashboard would immediately show unsaved changes. Updating the input here allows the diffing integration to update the panels in the session store before initializing the new dashboard. If we don't do this, the session store may have the old panel ids which causes the unsaved changes badge to appear. We don't want the user to save again as that would store the old panel ids in the new saved object.